astropy_timeseries/tests/test_downsample.py

```diff
@@ -0,0 +1,67 @@
+import pytest
+import numpy as np
+from numpy.testing import assert_equal
+
+from astropy import units as u
+from astropy.time import Time
+
+from ..sampled import TimeSeries
+from ..downsample import simple_downsample, reduceat
+
+INPUT_TIME = Time(['2016-03-22T12:30:31', '2016-03-22T12:30:32',
+                   '2016-03-22T12:30:33', '2016-03-22T12:30:34'])
+ts = TimeSeries(time=INPUT_TIME, data=[[1, 2, 3, 4]], names=['a'])
+ts_units = TimeSeries(time=INPUT_TIME, data=[[1, 2, 3, 4] * u.count], names=['a'])
+
+
+def test_reduceat():
+    # Similar to np.add for an array input.
+    output = reduceat(np.arange(8), [0, 4, 1, 5, 2, 6, 3, 7], np.sum)
+    # The result from np.add.reduceat is array([ 6,  4, 10,  5, 14,  6, 18,  7])
+    # A big difference.
+    assert_equal(output, np.array([6, 0, 10, 0, 14, 0, 18, 7]))
```

Comment:
So this is the difference between np.add.reduceat and np.sum with the current version of reduceat.

Reply:
Ok, we should fix our implementation to match numpy. The issue appears to be when the indices decrease.